packages/workspace/src/parser/parse.ts

```
@@ -48,3 +50,24 @@ export const parse = async (
     ? parseBufferAndFixErrors(srcString, filename, functions)
     : parseBuffer(srcString, filename, functions)
 }
+
+export type Token = {
+  value: string
+  type: string
+  col: number
+  line: number
+}
+
+// I don't return LexerToken because it would require the workspace package to
+// add @types/moo to the dependencies (instead of dev dependencies)
+export function *tokenizeContent(content: string): IterableIterator<Token> {
+  const lexer = new PeekableLexer(content)
+  try {
+    while (lexer.peek()) {
+      const token = lexer.next()
+      yield _.pick(token, ['value', 'type', 'col', 'line'])
+    }
+  // eslint-disable-next-line no-empty
```

why are we silencing the error in this case?
if we get an error here it means there is an invalid token in the content, right?
feels like we should at least have a log about it

---

Added a debug log. I didn't log it as an error since it is not such a rare case (e.g., the user is currently editing).